### PR TITLE
fix: no more test against SYSTEM user

### PR DESCRIPTION
**Pull request by martintamare**
### What does this PR do?
Remove hard test against the user for windows, so as to be correct with all windows language

### What issues does this PR fix or reference?
Related to #38246

The code is not factorized yet, I would like advice on the best practise to factorize the function get_sid_from_name.
How can we call other utils functions from the code.

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Review by cachedout — CHANGES_REQUESTED** [review]
I don't see an import for `win32security` or `pywintypes` anywhere. Those need to be added before this can work.

**Comment by martintamare** [comment]
Yep, agreed.
Any advice on how to factorize the get_sid_from_name ?
Can i cross call utils modules ? (don't know if that phrasing is correct in english ;)

File: salt/utils/win_runas.py [hunk 1]
```diff
@@ -392,10 +392,37 @@ def runas_system(cmd, username, password):
     return ret
 
 
+def get_sid_from_name(name):
```
Comment (cachedout) [inline salt/utils/win_runas.py]: To refactor this, just import `salt.utils` at the top of this file and then call `salt.utils.get_sid_from_name` where neccesary.

**Comment by martintamare** [comment]
Should be better now.
Only one thing I don't know : the get_user function in __init__.py
Should I remove the last else ?

File: salt/utils/__init__.py [hunk 2]
```diff
@@ -283,12 +289,10 @@ def get_user():
...
         return pwd.getpwuid(os.geteuid()).pw_name
+    elif HAS_WIN32:
+        return salt.utils.win_functions.get_current_user()
     else:
-        user_name = win32api.GetUserNameEx(win32api.NameSamCompatible)
-        if user_name[-1] == '$' and win32api.GetUserName() == 'SYSTEM':
-            # Make the system account easier to identify.
-            user_name = 'SYSTEM'
-        return user_name
+        return  # What should I do ?
```
Comment (cachedout) [inline salt/utils/__init__.py]: Here I would recommend that you `raise salt.exceptions.CommandExecutionError('Required external libraries not found. Need 'pwd' or 'win32api')`

**Comment by martintamare** [comment]
Code changed, and all commit merged into one.
We should be go to go.

**Comment by martintamare** [comment]
Hello there,
Any update on this ?

**Comment by cachedout** [comment]
@twangboy Can you please look at this?

**Comment by cachedout** [comment]
Go Go Jenkins!

File: salt/utils/__init__.py [hunk 3]
```diff
@@ -85,6 +85,12 @@
     HAS_WIN32API = False
 
 try:
+    import salt.utils.win_functions
+    HAS_WIN32 = True
```
Comment (twangboy) [inline salt/utils/__init__.py]: Here I would recommend `HAS_WIN32 = salt.utils.win_functions.HAS_WIN32` to make sure `win_functions` actually loads the win32 libraries.

**Review by twangboy — CHANGES_REQUESTED** [review]
I think this is a good fix to account for the different names of the `SYSTEM` account in different languages. Recommend for merge after requested changes are completed.

**Comment by martintamare** [comment]
Change applied !
Cheers,

**Review by twangboy — APPROVED** [review]

**Comment by cachedout** [comment]
Linted by hand and it passed. Since @twangboy signed off on this I am going to go ahead and merge it.

**Pull request merged**